scalameta/trees/src/main/scala/scala/meta/Trees.scala

```diff
@@ -373,7 +373,14 @@ object Defn {
                    templ: Template) extends Defn with Member.Type {
     // TODO: hardcoded in the @ast macro, find out a better way
     // require(templ.stats.getOrElse(Nil).forall(!_.is[Ctor]))
-    require(ctor.mods.isEmpty && ctor.paramss.isEmpty)
+
+    // TODO this doesn't work because the Dialect in implicit
```

Comment:
I don't think that we can have dialects in trees, unfortunately.

What if, for example, the tree is created manually, without any kind of parsing or quasiquoting? What dialect do we choose then?

However, what I think should be done is prohibiting trees like and types, or types, non-trivial trait constructors in contexts, where Scala211 is expected. For example, if you try to prettyprint such a tree in a Scala211 context, an exception must be thrown.

Wdyt?